aws-auth-cognito/src/main/java/com/amplifyframework/statemachine/codegen/states/AuthState.kt

```diff
@@ -23,32 +23,50 @@ import com.amplifyframework.statemachine.StateResolution
 import com.amplifyframework.statemachine.codegen.actions.AuthActions
 import com.amplifyframework.statemachine.codegen.data.AuthConfiguration
 import com.amplifyframework.statemachine.codegen.events.AuthEvent
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.Transient
 
+@Serializable
 sealed class AuthState : State {
+    @Serializable @SerialName("AuthState.NotConfigured")
```

Comment:
What is the need to append `AuthState`? I think since we already know the top level state is an AuthState, this is an extra overhead for serialization.

Reply:
This was added to differentiate `AuthState.NotConfigured` from `AuthenticationState.NotConfigured` or `Authorization..`.